src/wbase/FileChannelShared.cc

```diff
@@ -253,43 +253,13 @@ json FileChannelShared::filesToJson(vector<QueryId> const& queryIds, unsigned in
     return json::object({{"files", files}, {"num_selected", numSelected}, {"num_total", numTotal}});
 }
 
-shared_ptr<FileChannelShared> FileChannelShared::create(shared_ptr<wbase::SendChannel> const& sendChannel,
-                                                        qmeta::CzarId czarId, string const& workerId) {
+FileChannelShared::Ptr FileChannelShared::create(std::shared_ptr<wbase::UberJobData> const& uberJob) {
     lock_guard<mutex> const lock(_resultsDirCleanupMtx);
-    return shared_ptr<FileChannelShared>(new FileChannelShared(sendChannel, czarId, workerId));
+    return Ptr(new FileChannelShared(uberJob));
 }
 
-FileChannelShared::FileChannelShared(shared_ptr<wbase::SendChannel> const& sendChannel, qmeta::CzarId czarId,
-                                     string const& workerId)
-        : _sendChannel(sendChannel),
-          _uberJobId(0),
-          _czarId(czarId),
-          _czarHostName(""),  ///< Name of the czar host.
-          _czarPort(-1),
-          _workerId(workerId) {
-    LOGS(_log, LOG_LVL_DEBUG, "FileChannelShared created");
-    if (_sendChannel == nullptr) {
-        throw util::Bug(ERR_LOC, "FileChannelShared constructor given nullptr");
-    }
-}
-
-FileChannelShared::Ptr FileChannelShared::create(std::shared_ptr<wbase::UberJobData> const& uberJob,
-                                                 qmeta::CzarId czarId, string const& czarHostName,
-                                                 int czarPort, string const& workerId) {
-    lock_guard<mutex> const lock(_resultsDirCleanupMtx);
-    return Ptr(new FileChannelShared(uberJob, czarId, czarHostName, czarPort, workerId));
-}
-
-FileChannelShared::FileChannelShared(std::shared_ptr<wbase::UberJobData> const& uberJobData,
-                                     qmeta::CzarId czarId, string const& czarHostName, int czarPort,
-                                     string const& workerId)
-        : _sendChannel(nullptr),
-          _uberJobData(uberJobData),
-          _uberJobId(uberJobData->getUberJobId()),
-          _czarId(czarId),
-          _czarHostName(czarHostName),
-          _czarPort(czarPort),
-          _workerId(workerId) {
+FileChannelShared::FileChannelShared(std::shared_ptr<wbase::UberJobData> const& uberJobData)
+        : _sendChannel(nullptr), _uberJobData(uberJobData), _uberJobId(uberJobData->getUberJobId()) {
```

Comment:
You don't need to do this explicitly to the shared pointer variable:
```
_sendChannel(nullptr)
```
The default c-tor of the variable would do this for you.

Reply:
True, but including it indicates that it is intended and not an oversight.